jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/ContextHandlerLifeCycle.java

```diff
@@ -22,24 +22,25 @@
 
 import org.eclipse.jetty.deploy.graph.Graph;
 import org.eclipse.jetty.deploy.graph.Node;
+import org.eclipse.jetty.server.handler.ContextHandler;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * The lifecycle of an App in the {@link DeploymentManager}.
+ * The lifecycle of a {@link ContextHandler} in the {@link DeploymentManager}.
  * <p>
- * Setups a the default {@link Graph}, and manages the bindings to the life cycle via the {@link AppLifeCycle.Binding}
+ * Sets up the default {@link Graph}, and manages the bindings to the life cycle via the {@link ContextHandlerLifeCycle.Binding}
  * annotation.
  * <p>
- * <img alt="app lifecycle graph" src="doc-files/AppLifeCycle.png">
+ * <img alt="context-handler lifecycle graph" src="doc-files/ContextHandlerLifeCycle.png">
  */
-public class AppLifeCycle extends Graph
+public class ContextHandlerLifeCycle extends Graph
```

Comment:
I think this might be confusing with our normal `LifeCycle`.  Is this anyway a class that needs to be exposed, isn't it an implementation detail of the `DeploymentManager`?

Reply:
It's an important feature of the DeploymentManager, and is used by other implementations of providers.
And is also exposed via JMX, to allow controlling of the ContextHandlers through the ContextHandlerLifeCycle of the DeploymentManager.

Reply:
Where are there "providers" that need to reference this class? Pointing to a goal name defined in this class doesn't count.

Reply:
`ContextHandlerStates`?